Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,714 @@
+
+open CompactlySupported
+
+section
+
+/-- `CompactlySupportedContinuousMapClass F α β` states that `F` is a type of continuous maps with
+compact support.
+
+You should also extend this typeclass when you extend `CompactlySupportedContinuousMap`. -/
+class CompactlySupportedContinuousMapClass (F : Type*) (α β : outParam <| Type*)
+    [TopologicalSpace α] [Zero β] [TopologicalSpace β] [FunLike F α β]
+    extends ContinuousMapClass F α β : Prop where
+  /-- Each member of the class has compact support. -/
+hasCompactSupport (f : F) : HasCompactSupport f
+
+end
+
+export CompactlySupportedContinuousMapClass (hasCompactSupport)
+
+namespace CompactlySupportedContinuousMap
+
+section Basics
+
+variable [TopologicalSpace β] [Zero β] [FunLike F α β] [CompactlySupportedContinuousMapClass F α β]
+
+instance instFunLike : FunLike C_c(α, β) α β where
+  coe f := f.toFun
+  coe_injective' f g h := by
+    obtain ⟨⟨_, _⟩, _⟩ := f
+    obtain ⟨⟨_, _⟩, _⟩ := g
+    congr
+
+instance instCompactlySupportedContinuousMapClass :
+    CompactlySupportedContinuousMapClass C_c(α, β) α β where
+  map_continuous f := f.continuous_toFun
+  hasCompactSupport f := f.hasCompactSupport'
+
+instance instCoeTC : CoeTC F C_c(α, β) :=
+  ⟨fun f =>
+    { toFun := f
+      continuous_toFun := map_continuous f
+      hasCompactSupport' := hasCompactSupport f }⟩
+
+@[simp]
+theorem coe_toContinuousMap (f : C_c(α, β)) : (f.toContinuousMap : α → β) = f :=
+  rfl
+
+@[ext]
+theorem ext {f g : C_c(α, β)} (h : ∀ x, f x = g x) : f = g :=
+  DFunLike.ext _ _ h
+
+@[simp]
+theorem coe_mk (f : C(α, β)) (h : HasCompactSupport f) : ⇑(⟨f, h⟩ : C_c(α, β)) = f :=
+  rfl
+
+/-- Copy of a `CompactlySupportedContinuousMap` with a new `toFun` equal to the old one. Useful
+to fix definitional equalities. -/
+protected def copy (f : C_c(α, β)) (f' : α → β) (h : f' = f) : C_c(α, β) where
+  toFun := f'
+  continuous_toFun := by
+    rw [h]
+    exact f.continuous_toFun
+  hasCompactSupport' := by
+    simp_rw [h]
+    exact f.hasCompactSupport'
+
+@[simp]
+theorem coe_copy (f : C_c(α, β)) (f' : α → β) (h : f' = f) : ⇑(f.copy f' h) = f' :=
+  rfl
+
+theorem copy_eq (f : C_c(α, β)) (f' : α → β) (h : f' = f) : f.copy f' h = f :=
+  DFunLike.ext' h
+
+theorem eq_of_empty [IsEmpty α] (f g : C_c(α, β)) : f = g :=
+  ext <| IsEmpty.elim ‹_›
+
+/-- A continuous function on a compact space has automatically compact support. -/
+@[simps]
+def ContinuousMap.liftCompactlySupported [CompactSpace α] : C(α, β) ≃ C_c(α, β) where
+  toFun f :=
+    { toFun := f
+      continuous_toFun := f.continuous
+      hasCompactSupport' := hasCompactSupport_of_compactSpace f
+        }
+  invFun f := f
+  left_inv _ := rfl
+  right_inv _ := rfl
+
+/-- A continuous function on a compact space has automatically compact support. This is not an
+instance to avoid type class loops. -/
+lemma compactlySupportedContinuousMapClass.ofCompact {G : Type*} [FunLike G α β]
```

Comment:
```suggestion
lemma _root_.compactlySupportedContinuousMapClass.ofCompact (G : Type*) [FunLike G α β]
```
To avoid the full name `CompactlySupportedContinuousMap.compactlySupportedContinuousMapClass.ofCompact`. Also, make `G` explicit as there is no way to guess it.